lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/DriverStartHandler.java

```diff
@@ -73,10 +106,20 @@ public void onNext(final StartTime startTime) {
   }
 
   private void onRestart(final StartTime startTime) {
-    if (restartHandlers.isPresent()) {
-      for (EventHandler<StartTime> restartHandler : this.restartHandlers.get()) {
+    if (this.driverRestartManager.isPresent() && this.restartHandlers.isPresent()) {
+      for (EventHandler<StartTime> serviceRestartHandler : this.serviceRestartHandlers.get()) {
+        serviceRestartHandler.onNext(startTime);
+      }
+
+      for (EventHandler<StartTime> restartHandler : this.restartHandlers.get()){
         restartHandler.onNext(startTime);
       }
+
+      // This can only be called after calling client restart handlers because REEF.NET
+      // JobDriver requires making this call to set up the InterOp handlers.
+      final RestartEvaluatorInfo restartEvaluatorInfo = this.driverRestartManager.get().onRestartRecoverEvaluators();
```

Comment:
This seems a bit odd, as the return value of `onRestartRecoverEvaluators` is passed immediately to methods of the same instance of `RestartEvaluatorInfo`. Couldn't this be hidden behind a single method, `onDriverRestart()` or such?